spec/javascripts/components/intervention-spec.js

```diff
@@ -17,7 +47,16 @@ describe('The intervention component', function () {
     }
   })
 
-  it('triggers a tracking event when the component is displayed', function () {
-    expect(GOVUK.analytics.trackEvent).toHaveBeenCalledWith('interventionBanner', 'interventionShown')
+  it('triggers a tracking event when the suggested link is clicked', function () {
+    suggestionLink.click()
+    expect(GOVUK.analytics.trackEvent)
+      .toHaveBeenCalledWith('interventionBanner', 'interventionClicked')
```

Comment:
I don't think this test (and the one after it) is working as intended. When the user clicks this link, it should call `trackEvent`, but it should be calling it with more than the two attributes given (`'interventionBanner', 'interventionClicked'`).

You're using the `gem-track-click` module in the code to fire these events, but in this test environment that module isn't being initialised, so clicking on links won't fire those tracking events. However, this test is passing.

What's actually happening is the component is starting and firing the tracking events that this test is expecting from the `init` function, and that's what the test is picking up and passing with. To prove this, you can remove lines 51 and 57 - and the tests will still pass, even though nothing has been clicked. You can also combine each of the `expect` lines from each test into a single test, and it will still pass. Jasmine isn't exclusive about what it expects - if two calls to `trackEvent` are made and your test has only one expect in it that matches either of those calls, the test passes.

Recommendations:

- add a new test to check that events fired from the `init` function are working (all this needs to be is a test containing expects for those two events)
- don't use the `gem-track-click` module in the component for doing analytics, as it's almost impossible to test in this environment without doing something convoluted. Instead, set up click events to fire the tracking in the intervention component JS
- expand the `expect` lines in these two tests to include the category, action, label etc. that should be included with these events

If this is confusing, give me a shout and I can explain more.